rules/internal/utils/common_lints.go

```diff
@@ -71,6 +83,36 @@ func LintFieldMask(f *desc.FieldDescriptor) []lint.Problem {
 	return nil
 }
 
+// LintFieldProperties returns a problem if a message does not have the given singular-string
+// field or if that field is in a oneof.
+func LintFieldProperties(field, typName string, typ *builder.FieldType, wantOneof, wantSingular bool) func(*desc.MessageDescriptor) []lint.Problem {
```

Comment:
I would just write this function as `LintOneofField` and then expose `lintSingularField` as `LintSingularField`. Then, where this function is used, use each of these functions inline.

Mainly, this is because `LintFieldProperties` is generic and it would be clearer at the callsite that consumes this function to just call the two distinct functions than to pass in `true, true`, etc for `wantOneof` or `wantSinglular`.

Reply:
Ok, I agree. It was kind of a convenience thing to put them together and then pass the single function to the liner.Rule, but you're right your suggestion is more readable.

Reply:
Done! Broke use of `LintFieldProperties` into the three checks it encompassed.